test/rust_src/src/buffers-tests.el

```diff
@@ -104,7 +122,33 @@
           ;; and less than 1_000_000.
           (random-number (string-to-number (substring random-name (1+ (length buf-name))))))
       (should-not (string= random-name buf-name))
-      (should (< 0 random-number 999999)))))
+      (should (< 0 random-number 999999)))
+    (kill-buffer buf-name)))
+
+;; If we don't create any new buffers, the "other" buffer is *Messages*.
+(ert-deftest test-other-buffer-messages ()
+  (let* ((other (other-buffer)))
+    (should (string= (buffer-name other) "*Messages*"))))
+
+;; If we don't allow *Messages*, we end up with "*scratch*"
+(ert-deftest test-other-buffer-scratch ()
+  ;; *scratch* already exists, force recreation.
+  (kill-buffer "*scratch*")
+  (let* ((buf (get-buffer "*Messages*"))
+         (other (other-buffer buf)))
+    (should (string= (buffer-name other) "*scratch*"))))
+
+;; If we create several buffers, we end up with the most recent.
+(ert-deftest test-other-buffer-many ()
+  (let* ((msgs (get-buffer "*Messages*"))
+         (buf1 (get-buffer-create "1"))
+         (buf2 (get-buffer-create "2"))
+         (buf3 (get-buffer-create "3"))
+         (other (other-buffer msgs)))
+    (should (string= (buffer-name other) "1"))
+    (kill-buffer buf1)
+    (kill-buffer buf2)
+    (kill-buffer buf3)))
```

Comment:
I'm not quite sure how much testing to throw in. Fully testing `other-buffer`'s functionality would require writing a fair chunk of tests (exercising the predicates, visibility, frame-local buffer lists, etc.), and my elisp knowledge is pretty weak. If you think it's worth it, I can add more, but I don't want to create a huge pile of tests for a single function if that looks silly.

Reply:
Are there tests in the expensive suite? The more tests we have the more flexibility we have to experiment with editor internals.

Reply:
It is ok to add simple ones and open an Issue for "feature X needs more testing".

Reply:
I couldn't see any further tests of other-buffer. I've added a test for the predicate, but the part with visibility requires much more understanding of how to test frames/windows than I currently have.